core/include/scipp/core/transform.h

```diff
@@ -235,10 +235,51 @@ struct is_eigen_type<sparse_container<Eigen::Matrix<T, Rows, Cols>>>
 template <class T>
 inline constexpr bool is_eigen_type_v = is_eigen_type<T>::value;
 
+namespace transform_detail {
+template <class T> struct is_sparse : std::false_type {};
+template <class T> struct is_sparse<sparse_container<T>> : std::true_type {};
+template <class T>
+struct is_sparse<ValuesAndVariances<sparse_container<T>>> : std::true_type {};
+template <class T>
+struct is_sparse<ValuesAndVariances<const sparse_container<T>>>
+    : std::true_type {};
+template <class T> inline constexpr bool is_sparse_v = is_sparse<T>::value;
+} // namespace transform_detail
+
+template <class T> auto check_and_get_size(const T &a) {
+  return scipp::size(a);
+}
+
+template <class T1, class T2>
+auto check_and_get_size(const T1 &a, const T2 &b) {
+  if constexpr (transform_detail::is_sparse_v<T1>) {
+    if constexpr (transform_detail::is_sparse_v<T2>)
+      expect::sizeMatches(a, b);
+    return scipp::size(a);
+  } else {
+    return scipp::size(b);
```

Comment:
What is the significance of returning `size b`?

Reply:
The function is actually used when implementing the operation (just reusing the some function in the size check). This is required to decide how to iterate in the different cases of operations (scalar with scalar, scalar with sparse container, sparse container with scalar, ...).